configure

```diff
@@ -2185,6 +2185,10 @@ ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
+# Check enviroment virable ENABLE_PRINTF
+if [ "$ENABLE_PRINTF" = "ON" ]; then
+    CXXFLAGS+=" -DENABLE_PRINTF=ON "
+fi
```

Comment:
configure file is an auto-generated file, upstream files should be updated as well.

Reply:
> configure file is an auto-generated file, upstream files should be updated as well.

It is not an auto-generated file, run this script to generate makefile

Reply:
> It is not an auto-generated file, run this script to generate makefile

configure is auto-generated by configure.ac, you can ref to GNU autoconf manual for detail.

Reply:
> > It is not an auto-generated file, run this script to generate makefile
> 
> configure is auto-generated from configure.ac, you can ref to GNU autoconf manual for detail.

I compared the autoconf genarated file with configure, the authour make many changes in configure file, so I think it's ok to do some change in configure file not in cofigure.ac